### PR TITLE
feat: Lint other filetypes with prettier

### PR DESCRIPTION
**Pull request by vikr01**
`prettier` supports linting for [various other languages besides javascript](https://github.com/prettier/prettier#opinionated-code-formatter).

- Use prettier to autofix `json`, `md`, `html`, `yml`, and `css` files since they're all used in this repo
  - Used during precommit
  - Used in `postlint:fix` script that occurs after the `lint:fix` script
- Change `.prettierrc` to `.prettierrc.json` so it can be linted (could also keep it as `.prettierrc` and use the `overrides` option in the config)
- Include dotfiles in `lint-staged` since they aren't be default
- Don't lint `package.json` with `prettier` since it gets changed by package managers.

### CODE REVIEW

**Comment by Hartmarken348** [comment]
Feat

File: packages/@pollyjs/adapter-fetch/CHANGELOG.md [hunk 1]
```diff
@@ -7,115 +7,80 @@ See [Conventional Commits](https://conventionalcommits.org) for commit guideline
 
 **Note:** Version bump only for package @pollyjs/adapter-fetch
```
Comment (offirgolan) [inline packages/@pollyjs/adapter-fetch/CHANGELOG.md]: All `CHANGELOG.md` files should be ignored from linting as they are auto generated.

File: packages/@pollyjs/ember/tests/index.html [hunk 2]
```diff
@@ -1,33 +1,29 @@
 <!DOCTYPE html>
```
Comment (offirgolan) [inline packages/@pollyjs/ember/tests/index.html]: Auto generated by ember-cli. This should be ignored from linting.

File: packages/@pollyjs/ember/tests/dummy/app/index.html [hunk 3]
```diff
@@ -1,16 +1,16 @@
 <!DOCTYPE html>
```
Comment (offirgolan) [inline packages/@pollyjs/ember/tests/dummy/app/index.html]: Auto generated by ember-cli. This should be ignored from linting.

File: packages/@pollyjs/adapter-fetch/CHANGELOG.md (hunk 1 above)
Comment (vikr01) [reply packages/@pollyjs/adapter-fetch/CHANGELOG.md]: Done.

File: packages/@pollyjs/ember/tests/index.html (hunk 2 above)
Comment (vikr01) [reply packages/@pollyjs/ember/tests/index.html]: Done.

**Pull request merged**